INSTALL.md

```diff
@@ -1 +1,114 @@
-# TODO
+# Building opentelemetry-cpp
+
+[Bazel](https://bazel.build) and [CMake](https://cmake.org/) are the official
+build systems for opentelemetry-cpp.
+
+## Build instructions using CMake
+
+### Prerequisites
+
+- A supported platform (eg Windows, macOS, Linux etc).
+Refer [Platforms Supported](./README.md#supported-development-platforms)
+for more information.
+- A compatible C++ compiler supporting atleast C++11.
+Major compilers are supported.
+Refer [Supported Compilers](./README.md#supported-c-versions) for more information.
+- [Git](https://git-scm.com/) for fetching opentelemetry-cpp source code from repository.
+To install Git, consult the [Set up Git](https://help.github.com/articles/set-up-git/)
+guide on GitHub.
+- [CMake](https://cmake.org/) for building opentelemetry-cpp API,
+SDK with their unittests. We use CMake version 3.15.2 in our build system.
+To install CMake, consult the [Installing CMake](https://cmake.org/install/) guide.
+- [GoogleTest](https://github.com/google/googletest) framework to build and
+run the unittests.
+We use GoogleTest version 1.10.0 in our build system. To install GoogleTest,
+consult the [GoogleTest Build Instructions](https://github.com/google/googletest/blob/master/googletest/README.md#generic-build-instructions).
+- Apart from above core requirements, the Exporters and Propagators have their
+build dependencies which are not covered here. E.g, Otlp Exporter needs
+grpc/protobuf library, Zipkin exporter needs nlohmann-json and libcurl,
+ETW exporter need nlohmann-json to build. This is covered in the build
+instructions for these components.
+
+### Building as Standalone CMake Project
+
+1. Getting the opentelementry-cpp source:
+
+   ```console
+   # Change to the directory where you want to create the code repository
+   $ cd ~
+   $ mkdir source && cd source
+   $ git clone https://github.com/open-telemetry/opentelemetry-cpp
+   Cloning into 'opentelemetry-cpp'...
+   ...
+   Resolving deltas: 100% (3225/3225), done.
+   $
+   ```
+
+2. Navigate to the repository cloned above, and create the `CMake`
+build configuration.
+
+   ```console
+   $ cd opentelemetry-cpp
+   $ mkdir build && cd build
+   $ cmake .. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS="-Werror"
```

Comment:
What do you think about adding a section to enable position independent code on supported compilers? otel-cpp does not enable it by default, might be useful for someone (would have been for me when I started using it) to add a note about enabling it, i.e. `cmake .. -DCMAKE_POSITION_INDEPENDENT_CODE=ON`

And perhaps a note about enabling OTLP: `-DWITH_OTLP=ON`

Reply:
Good comment. Have updated the document accordingly.